scripts/tf-tests/run-jobs.py

```diff
@@ -0,0 +1,219 @@
+    )
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--send-jobs",
+        action="store_true",
+        help="Only submit jobs without monitoring their statuses",
+    )
+    group.add_argument(
+        "--check-status",
+        action="store_true",
+        help="Only check job statuses without submitting new jobs",
+    )
+    return parser.parse_args()
+
+
+# Job Submission Functions
+def load_canonical_ids(filename: Path) -> list:
+    """Reads the Canonical IDs from a file."""
+    with open(filename, "r", encoding="utf8") as file:
+        return file.read().strip().splitlines()
+
+
+def create_job_yaml(template_file: Path, canonical_id: str) -> str:
+    """Creates a modified job YAML for a specific Canonical ID."""
+    with open(template_file, "r", encoding="utf8") as file:
+        job_yaml = file.read()
+    return job_yaml.replace("$CANONICAL_ID", canonical_id)
+
+
+def write_temp_job_file(job_yaml: str, output_dir: Path, canonical_id: str) -> Path:
+    """Writes the modified job YAML to a temporary file."""
+    temp_job_file = output_dir / f"{canonical_id}_tf-job.yaml"
+    temp_job_file.write_text(job_yaml)
+    return temp_job_file
+
+
+def submit_job(temp_job_file: Path, canonical_id: str) -> Optional[str]:
+    """Submits the job and returns the job UUID."""
+    try:
+        result = subprocess.run(
+            ["testflinger", "submit", str(temp_job_file)],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        for line in result.stdout.splitlines():
+            if line.startswith("job_id:"):
+                return line.split(": ")[1].strip()
+        logging.warning("Failed to retrieve job_id for %s", canonical_id)
+    except subprocess.CalledProcessError as e:
+        logging.error("Error submitting job for %s: %s", canonical_id, e.stderr)
+    return None
+
+
+def save_job_uuid(job_uuid: str, output_dir: Path, canonical_id: str):
+    """Creates a directory for the Canonical ID and saves the job UUID."""
+    id_dir = output_dir / canonical_id
+    id_dir.mkdir(exist_ok=True)
+    (id_dir / "tf_job_id.txt").write_text(job_uuid)
+    logging.info("Job submitted for %s with job_id: %s", canonical_id, job_uuid)
+
+
+def submit_all_jobs(
+    machines_file: Path, template_file: Path, output_dir: Path
+) -> Dict[str, Path]:
+    """Submit all jobs for the given machines."""
+    canonical_ids = load_canonical_ids(machines_file)
+    job_ids = {}
+
+    for canonical_id in canonical_ids:
+        job_yaml = create_job_yaml(template_file, canonical_id)
+        temp_job_file = write_temp_job_file(job_yaml, output_dir, canonical_id)
+        job_uuid = submit_job(temp_job_file, canonical_id)
+
+        if job_uuid:
+            save_job_uuid(job_uuid, output_dir, canonical_id)
+            job_ids[job_uuid] = output_dir / canonical_id
+
+        temp_job_file.unlink()  # Clean up temporary YAML file
```

Comment:
Up to you if you use it or not, but whenever there's this pattern of creating and deleting temporary files, you can also use the [contexts in `tempfile`](https://docs.python.org/3.8/library/tempfile.html).